### PR TITLE
opcm-redesign: op-deployer changes

### PR DESCRIPTION
**Pull request by blmalone**
This is a stacked PR that works with: https://github.com/ethereum-optimism/optimism/pull/12851 

This Pr introduces the necessary go changes to work with the new OPCM architecture.

### CODE REVIEW

**Comment by blmalone** [comment]
* **#12904** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12904?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12904?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#12851** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12851?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: op-deployer/pkg/deployer/pipeline/alt_da.go [hunk 1]
```diff
@@ -27,11 +27,16 @@ func DeployAltDA(env *Env, intent *state.Intent, st *state.State, chainID common
...
+	if err != nil {
+		return err
+	}
+
 	var dao opcm.DeployAltDAOutput
 	lgr.Info("deploying alt-da contracts")
 	dao, err = opcm.DeployAltDA(env.L1ScriptHost, opcm.DeployAltDAInput{
 		Salt:                     st.Create2Salt,
-		ProxyAdmin:               st.ImplementationsDeployment.OpcmProxyAddress,
+		ProxyAdmin:               st.Chains[chainIndex].ProxyAdminAddress, // TODO: @blmalone explain this change to slipper.
```
Comment (blmalone) [inline op-deployer/pkg/deployer/pipeline/alt_da.go]: @mslipper I think this was a bug. Let me know if this fix looks good to you.

File: op-deployer/pkg/deployer/standard/standard.go [hunk 2]
```diff
@@ -134,10 +134,11 @@ func ManagerImplementationAddrFor(chainID uint64) (common.Address, error) {
 	switch chainID {
 	case 1:
 		// Generated using the bootstrap command on 10/18/2024.
-		return common.HexToAddress("0x18cec91779995ad14c880e4095456b9147160790"), nil
+		// TODO: @blmalone this needs re-bootstrapped because it's still proxied
+		return common.HexToAddress(""), nil
```
Comment (blmalone) [inline op-deployer/pkg/deployer/standard/standard.go:138]: @blmalone Still need to deploy mainnet opcm.

File: op-deployer/pkg/deployer/standard/standard.go [hunk 3]
```diff
@@ -134,10 +134,11 @@ func ManagerImplementationAddrFor(chainID uint64) (common.Address, error) {
...
 	case 11155111:
-		// Generated using the bootstrap command on 10/18/2024.
-		return common.HexToAddress("0xf564eea7960ea244bfebcbbb17858748606147bf"), nil
+		// Generated using the bootstrap command on 11/13/2024.
+		return common.HexToAddress("0xdb78ead76e76af1f552b96fa262e6a64d8e95c30"), nil
```
Comment (blmalone) [inline op-deployer/pkg/deployer/standard/standard.go]: This has a `l1ContractsReleaseVersion` value of `dev`.

File: op-deployer/pkg/deployer/pipeline/alt_da.go [hunk 4]
```diff
@@ -27,11 +27,16 @@ func DeployAltDA(env *Env, intent *state.Intent, st *state.State, chainID common
 		return nil
 	}
 
+	chainIndex, err := findChainIndex(st.Chains, chainID)
```
Comment (mslipper) [inline op-deployer/pkg/deployer/pipeline/alt_da.go]: [nit] There's already a util for this - `st.StateFor(chainID)`

File: op-deployer/pkg/deployer/pipeline/alt_da.go [hunk 5]
```diff
@@ -47,6 +52,15 @@ func DeployAltDA(env *Env, intent *state.Intent, st *state.State, chainID common
 	return nil
 }
 
+func findChainIndex(chains []*state.ChainState, chainID common.Hash) (int, error) {
```
Comment (mslipper) [inline op-deployer/pkg/deployer/pipeline/alt_da.go]: you can delete this method, see above

File: op-deployer/pkg/deployer/pipeline/alt_da.go (hunk 1 above)
Comment (mslipper) [reply op-deployer/pkg/deployer/pipeline/alt_da.go]: yes that looks like a bug, good catch

File: op-deployer/pkg/deployer/pipeline/alt_da.go (hunk 5 above)
Comment (blmalone) [reply op-deployer/pkg/deployer/pipeline/alt_da.go]: https://github.com/ethereum-optimism/optimism/pull/12904/commits/ccd954ebc5c4d06da92537994c6abee9ad08666a

File: op-deployer/pkg/deployer/pipeline/alt_da.go (hunk 4 above)
Comment (blmalone) [reply op-deployer/pkg/deployer/pipeline/alt_da.go]: https://github.com/ethereum-optimism/optimism/pull/12904/commits/ccd954ebc5c4d06da92537994c6abee9ad08666a

**Pull request merged**